### PR TITLE
Carbons Can Now Special Click on Window Frame Weeds to Climb Frames

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Weeds no longer obstruct special click from climbing window frames.

## Why It's Good For The Game

Weeds no longer obstruct special click from climbing window frames.

## Changelog
:cl:
code: Weeds no longer obstruct special click from climbing window frames.
/:cl:

### CODE REVIEW

**Review by TiviPlus — CHANGES_REQUESTED** [review]
I told you to use signals

**Comment by Surrealaser** [comment]
As ever, fuck off linters

File: code/datums/keybinding/carbon.dm [hunk 1]
```diff
@@ -146,3 +146,4 @@
 		UnregisterSignal(user, (COMSIG_MOB_CLICKON))
 		return
 	A.specialclick(user)
+	SEND_SIGNAL(A, COMSIG_MOB_SPECIAL_CLICK, user)
```
Comment (TiviPlus) [inline code/datums/keybinding/carbon.dm:149]: Put it inside specialclick

File: code/game/objects/effects/weeds.dm [hunk 2]
```diff
@@ -110,6 +110,10 @@
 /obj/effect/alien/weeds/weedwall/window
 	layer = ABOVE_TABLE_LAYER
 
+/obj/effect/alien/weeds/weedwall/window/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_MOB_SPECIAL_CLICK, .proc/special_click_signal)
+
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:116]: Register on init for the window under instead

File: code/game/objects/effects/weeds.dm [hunk 3]
```diff
@@ -121,9 +125,20 @@
...
+	if(!F)
+		return
+	return INVOKE_ASYNC(F, /obj/structure/window_frame/proc.specialclick, user)
+
 /obj/effect/alien/weeds/weedwall/frame
 	layer = ABOVE_TABLE_LAYER
 
+/obj/effect/alien/weeds/weedwall/frame/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_MOB_SPECIAL_CLICK, .proc/special_click_signal)
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:140]: ditto
also why is this not on /weedwall?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/weeds.dm (hunk 2 above)
Comment (Surrealaser) [reply code/game/objects/effects/weeds.dm]: This doesn't seem to work; the atom you click is the weeds, which then fires the signal at the weeds.

Trying to then do say RegisterSignal(window, COMSIG_MOB_SPECIAL_CLICK, obj/structure/window_frame.proc/climb_or_whatever) on initialization goes runtime. If you really want, I'll tool around with it some more, maybe there was something missing, but I didn't have too much time to dick around and I figured the alternate solution I produced was fine

**Pull request closed without merging**